[TypeChecker] Enable result builder AST transform by default

xedin opened the pull request:
This PR:

- Drops experimental feature flag (`ResultBuilderASTTransform`)
- Adjusts code completion (changes by @ahoppen)
- Adjusts all the improved test-cases
- Removes original result builder implementation

Resolves: rdar://89880662
Resolves: rdar://97388257
Resolves: rdar://95235303
Resolves: rdar://96343005
Resolves: rdar://91731420
Resolves: rdar://95977761
Resolves: rdar://80680636


<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

ahoppen committed:
[CodeCompletion] Insert code completion token into build block call to retrieve contextual type

rdar://94224692

ahoppen committed:
[CodeCompletion] Drop ForCodeCompletion constraint system option for syntactic elements that don't contain completion token

xedin committed:
[ResultBuilder] Don't run syntactic diagnostics on transformed do statements

Visiting of transformed `do` shouldn't run syntatic diagnostics because
they are going to be performed by `visit` that called `visitDoStmt`.

xedin committed:
[TypeChecker] Enable result builder AST transform by default

xedin committed:
[AST] Drop `ResultBuilderASTTransform` feature flag

xedin committed:
[TypeChecker] NFC: Remove original result builder implementation

LucianoPAlmeida commented on include/swift/Sema/ConstraintSystem.h:
```diff
@@ -6245,6 +6245,10 @@ class ConjunctionElement {
   }
 
+  /// Returns \c false if this conjunction element is known not to contain the
+  /// code compleiton token.
```
Typo?
```suggestion
  /// code completion token.
```

xedin replied:
Thanks!

xedin commented:
@swift-ci please test

xedin commented:
@swift-ci please test

xedin commented:
@swift-ci please test macOS platform

xedin commented:
@swift-ci please test macOS platform

xedin committed:
[BuilderTransform] Fix a couple post-rebase API mismatches

- `buildCallIfNeeded` has been replaced with `buildCall`
- SolverState has a "current indent" method now.

xedin commented:
@swift-ci please test macOS platform

xedin commented:
@swift-ci please test

xedin commented:
@swift-ci please build toolchain macOS platform

xedin commented:
@swift-ci please build toolchain macOS platform

xedin commented:
@swift-ci please smoke test

xedin committed:
[BuilderTransform] NFC: Adjust variable declaration tests

With AST transform enabled by default most of the previously
invalid declarations became valid (due to SE-0373 being implemented
only for AST transform).

xedin commented:
@swift-ci please smoke test macOS platform

xedin commented:
@swift-ci please smoke test

xedin commented:
@swift-ci please smoke test

xedin commented:
@swift-ci please test
